src/wallet/api.rs

```diff
@@ -1004,40 +1019,169 @@ impl Wallet {
         let mut swap_coin_utxo = self.list_incoming_swap_coin_utxo_spend_info(Some(&all_utxos))?;
         seed_coin_utxo.append(&mut swap_coin_utxo);
 
-        // Fetch utxos, filter out existing fidelity coins
-        let mut unspents = seed_coin_utxo
+        // Filter out existing fidelity coins
+        let unspents: Vec<_> = seed_coin_utxo
             .into_iter()
             .filter(|(_, spend_info)| !matches!(spend_info, UTXOSpendInfo::FidelityBondCoin { .. }))
-            .collect::<Vec<_>>();
+            .collect();
 
-        unspents.sort_by(|a, b| b.0.amount.cmp(&a.0.amount));
+        // Construct TxIn from unspents
+        let inputs: Vec<TxIn> = unspents
+            .iter()
+            .map(|(utxo, _)| TxIn {
+                previous_output: OutPoint {
+                    txid: utxo.txid,
+                    vout: utxo.vout,
+                },
+                script_sig: utxo.script_pub_key.clone(),
+                sequence: Sequence::MAX, // Default sequence number
+                witness: utxo
+                    .witness_script
+                    .as_ref()
+                    .map_or_else(Witness::default, |script| {
+                        let mut witness = Witness::new();
+                        witness.push(script.as_bytes());
+                        witness
+                    }),
+            })
+            .collect();
 
-        let mut selected_utxo = Vec::new();
-        let mut remaining = amount;
+        // Get next indexes without modifying state
+        let next_external_index = self.find_hd_next_index(KeychainKind::External)?;
+        let next_internal_index = self.find_hd_next_index(KeychainKind::Internal)?;
 
-        // the simplest largest first coinselection.
-        for unspent in unspents {
-            if remaining.checked_sub(unspent.0.amount).is_none() {
-                selected_utxo.push(unspent);
-                break;
-            } else {
-                remaining -= unspent.0.amount;
-                selected_utxo.push(unspent);
+        // Get descriptors
+        let descriptors = self.get_wallet_descriptors()?;
+
+        // Get external address
+        let external_descriptor = descriptors
+            .get(&KeychainKind::External)
+            .expect("external keychain expected");
+        let external_address = self.rpc.derive_addresses(
+            external_descriptor,
+            Some([next_external_index, next_external_index]),
+        )?[0]
+            .clone()
+            .assume_checked();
+
+        // Get internal address
+        let internal_descriptor = descriptors
+            .get(&KeychainKind::Internal)
+            .expect("internal keychain expected");
+        let internal_address = self.rpc.derive_addresses(
+            internal_descriptor,
+            Some([next_internal_index, next_internal_index]),
+        )?[0]
+            .clone()
+            .assume_checked();
```

Comment:
There are dediacted wallet APIs to get the next internal and external addresses. You can just use that?